### PR TITLE
MEN-2640: mender-convert does not add demo cert to mender.conf

### PR DESCRIPTION
**Pull request by mirzak**
Supersedes #104

### CODE REVIEW

**Comment by mirzak** [comment]
@enurseitov I slight modified your PR. I have added your sign-off tag to the second commit and I will need an ack from you that this is ok.

**Comment by enurseitov** [comment]
@mirzak sure, thanks for taking this over, I was blocked with my initial PR, didn't manage to make the pipelines accept my commits.

Feel free to delete the original PR

File: convert-stage-4.sh [hunk 1]
```diff
@@ -193,17 +193,20 @@ install_files() {
...
-  # Install provided or demo certificate
+  # Install provided
   if [ -n "${server_cert}" ]; then
     sudo install -m 0444 ${server_cert} ${primary_dir}/${sysconfdir}/server.crt
-  else
-    sudo install -m 0444 ${mender_dir}/server.demo.crt ${primary_dir}/${sysconfdir}/server.crt
+  fi
+
+  if [ -e "${primary_dir}/${sysconfdir}/server.crt" ]; then
+    jq_inplace '.ServerCertificate = \"$'{primary_dir}'/'${sysconfdir}'/server.crt\"' ${primary_dir}/${sysconfdir}/mender.conf
```
Comment (lluiscampos) [inline convert-stage-4.sh]: Careful, the dollar sign is before the single quote. It should be:

```
jq_inplace '.ServerCertificate = \"'${primary_dir}'/'${sysconfdir}'/server.crt\"' ${primary_dir}/${sysconfdir}/mender.conf
```

Comment (mirzak) [reply convert-stage-4.sh]: Thanks, fixed.

**Comment by mirzak** [comment]
Ah damn, was trigger happy with the merge. Now it resolves to,

```
  "ServerCertificate": "/mender-convert/output/sdimg/primary/etc/mender/server.crt"
```

So `${primary_dir}` should have been dropped completely from the that line and that explains why it "worked" earlier when there was a typo.

**Comment by lluiscampos** [comment]
Of course! It should only be `/${sysconfdir}`

Sorry :disappointed:

**Comment by mirzak** [comment]
Follow-up PR #109

**Pull request merged**